monai/transforms/intensity/array.py

```diff
@@ -907,6 +907,8 @@ def __call__(self, img: NdarrayOrTensor) -> NdarrayOrTensor:
             if self.divisor is not None and len(self.divisor) != len(img):
                 raise ValueError(f"img has {len(img)} channels, but divisor has {len(self.divisor)} components.")
 
+            img, *_ = convert_data_type(img, dtype=torch.float32)
```

Comment:
Thanks for the fix. Seems we can remove the convert out side of the `self._normalize` to avoid convert multiple times here.

Reply:
Should we also check that the dtype of `img` isn't float16/float32/float64 and only do the conversion in that case?

Reply:
Yes, it would be a little more rigorous. Although in the non channel-wise version, we do the conversion directly. But yes if we could do the check, would be better.

Reply:
```suggestion
            if not img.dtype.is_floating_point:
                img, *_ = convert_data_type(img, dtype=torch.float32)
```
Something like this?